site/public/js/submission-page.js

```diff
@@ -7,3 +8,124 @@ function openActionsPopup(popup_css, element_id) {
     my_window.document.close();
     my_window.focus();
 }
+
+let days, hours, mins, seconds = 0;
+let curTime = 0;
+let lastTime = 0;
+let deadline = 0;
+let user_deadline = 0;
+let startTime = 0;
+let width = 0;
+let allowedTime = 0;
+let gradeable_id = '';
+let ticks_till_update = 600000;
+
+function initializeTimer(gradeableID) {
+    gradeable_id = gradeableID;
+    syncWithServer();
+}
+
+function syncWithServer() {
+    // eslint-disable-next-line no-undef
+    const url = buildCourseUrl(['gradeable', gradeable_id, 'time_remaining_data']);
+    $.ajax({
+        url,
+        type: 'GET',
+        processData: false,
+        contentType: false,
+        success: function(res) {
+            lastTime = Date.now();
+            const response = JSON.parse(res);
+            ticks_till_update = 600000;
+            if (response.status === 'success') {
+                const { data } = response;
+                if (Object.prototype.hasOwnProperty.call(data, 'user_allowed_time_deadline')) {
+                    user_deadline = data.user_allowed_time_deadline;
+                }
+                if (Object.prototype.hasOwnProperty.call(data, 'user_allowed_time')) {
+                    allowedTime = data.user_allowed_time;
+                }
+                if (Object.prototype.hasOwnProperty.call(data, 'user_start_time')) {
+                    startTime = data.user_start_time;
+                }
+                curTime = data.current_time;
+                deadline = data.deadline;
+                updateTime();
+            }
+            else {
+                // eslint-disable-next-line no-undef
+                displayErrorMessage('Something went wrong while starting the timer');
+            }
+        },
+        error: function (err) {
+            console.log(err);
+        },
+    });
+}
+
+function updateTime() {
+    if (Math.abs(Date.now() - lastTime) > 5000 || ticks_till_update <= 0) {
+        //we need to sync back up
+        syncWithServer();
+    }
+    else {
+        curTime += (Date.now()-lastTime);
+        ticks_till_update -= (Date.now()-lastTime);
+        lastTime = Date.now();
+        if (document.getElementById('gradeable-time-remaining-text') !== null) {
```

Comment:
Any reason why you're using `document.getElementById()` instead of JQuery here?  (same thing for this whole function)

Reply:
There isn't necessarily a specific reason but there isn't much advantage or need to do so for this since I am simply getting elements by id with normal JavaScript.